src/liballoc/collections/btree/map.rs

```diff
@@ -1606,6 +1641,76 @@ impl<K, V> Clone for Values<'_, K, V> {
     }
 }
 
+/// An iterator produced by calling `drain_filter` on BTreeMap.
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+pub struct DrainFilter<'a, K, V, F>
+where
+    K: 'a + Ord,
+    V: 'a,
+    F: 'a + FnMut(&K, &mut V) -> bool,
+{
+    pred: F,
+    retained: usize,
+    length: &'a mut usize,
+    cur_leaf_edge: Handle<NodeRef<marker::Mut<'a>, K, V, marker::Leaf>, marker::Edge>,
+}
+
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+impl<'a, K, V, F> Drop for DrainFilter<'a, K, V, F>
+where
+    K: 'a + Ord,
+    V: 'a,
+    F: 'a + FnMut(&K, &mut V) -> bool,
+{
+    fn drop(&mut self) {
+        self.for_each(drop);
+    }
+}
+
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+impl<'a, K, V, F> fmt::Debug for DrainFilter<'a, K, V, F>
+where
+    K: 'a + fmt::Debug + Ord,
+    V: 'a + fmt::Debug,
+    F: 'a + FnMut(&K, &mut V) -> bool,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_tuple("DrainFilter").field(&self.length).finish()
```

Comment:
This was definitely one I "just make it compile"d and forgot all about.

Reply:
We don't really need a genuine iterator, right, we just need to iterate to the end (which is easy to do after #68827) and call `.entry` each time. But I wonder if that's ideal for debugging. I've used it for set intersection and such myself, and if you're debugging a large set, you're screwed. Ideally someone should implement a stop in the debug formatter after 50 entries or so, but they didn't. Instead of listing everything, just showing an Option with the next key (with or without value), seems fine to me.

`linked_list::Iter` also doesn't list elements, but just a counter.